datafusion/physical-expr/src/utils/guarantee.rs

```diff
@@ -120,7 +114,7 @@ impl LiteralGuarantee {
     /// expression is guaranteed to be `null` or `false`.
     ///
     /// # Notes:
-    /// 1. `expr` must be a boolean expression.
+    /// 1. `expr` must be a boolean expression or inlist expression.
```

Comment:
More specifically, the boolean expr can only be in a form that can be transformed into a `ColOpLit`